turtlebot2_drivers/src/kobuki_node.cpp

```diff
@@ -31,47 +48,44 @@ double g_max_vyaw;
 void cmdVelCallback(const geometry_msgs::msg::Twist::SharedPtr msg)
 {
   std::lock_guard<std::mutex> kobuki_guard(g_kobuki_mutex);
-  std::cout << "Received: (" << msg->linear.x << "," << msg->angular.z << ")" << std::endl;
   double vx = std::min(std::max(msg->linear.x, -g_max_vx), g_max_vx);
   double vyaw = std::min(std::max(msg->angular.z, -g_max_vyaw), g_max_vyaw);
   g_kobuki->setBaseControl(vx, vyaw);
   g_last_cmd_vel_time = std::chrono::system_clock::now();
 }
 
-void on_parameter_event(const rcl_interfaces::msg::ParameterEvent::SharedPtr event)
-{
-  // TODO: handle dynamic params
-  std::cout << "Parameter event:" << std::endl << " new parameters:" << std::endl;
-  for (auto & new_parameter : event->new_parameters) {
-    std::cout << "  " << new_parameter.name << std::endl;
-  }
-  std::cout << " changed parameters:" << std::endl;
-  for (auto & changed_parameter : event->changed_parameters) {
-    std::cout << "  " << changed_parameter.name << std::endl;
-  }
-  std::cout << " deleted parameters:" << std::endl;
-  for (auto & deleted_parameter : event->deleted_parameters) {
-    std::cout << "  " << deleted_parameter.name << std::endl;
-  }
-}
-
 int main(int argc, char * argv[])
 {
   rclcpp::init(argc, argv);
 
   auto node = rclcpp::node::Node::make_shared("kobuki_node");
   auto parameter_service = std::make_shared<rclcpp::parameter_service::ParameterService>(node);
   auto parameters_client = std::make_shared<rclcpp::parameter_client::SyncParametersClient>(node);
-  auto sub = parameters_client->on_parameter_event(on_parameter_event);
+  auto cmd_vel_sub = node->create_subscription<geometry_msgs::msg::Twist>(
+    "cmd_vel", cmdVelCallback, rmw_qos_profile_default);
+  auto odom_pub = node->create_publisher<nav_msgs::msg::Odometry>("odom", rmw_qos_profile_default);
+  auto imu_pub = node->create_publisher<sensor_msgs::msg::Imu>("imu_data", rmw_qos_profile_default);
+  tf2_ros::TransformBroadcaster br(node);
 
   kobuki::Parameters parameters;
   parameters.device_port = "/dev/kobuki";
   g_max_vx = 0.5;
   g_max_vyaw = 1.0;
-  // TODO use to-be-written params API to get a single value, with a default
+  std::string odom_frame = "odom";
+  std::string base_link_frame = "base_link";
+  std::string gyro_link_frame = "gyro_link";
   for (auto & parameter : parameters_client->get_parameters({"device_port"})) {
     parameters.device_port = parameter.as_string();
   }
+  for (auto & parameter : parameters_client->get_parameters({"odom_frame"})) {
```

Comment:
@wjwwood, I haven't seen the problem that you're describing, but if I understand correctly, I'm not going through the same sequence that you are in ros2/rclcpp#308. By luck, I've created all of my topics before actually using the `SyncParametersClient`.

But as you and @rohbotics have suggested, I should just use `Node::get_parameter` instead; I'll do that. (Was that interface always available? I wonder why I went for the client/server setup...)